lib/rules/newline-after-var.js

```diff
@@ -11,35 +12,70 @@
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Determine if provided keyword is a variable declaration
+     * @private
+     * @param {String} keyword - keyword to test
+     * @returns {Boolean} True if `keyword` is a type of var
+     */
+    function isVar(keyword) {
+        return keyword === "var" || keyword === "let" || keyword === "const";
     }
 
-    return {
-        "VariableDeclaration:exit": function(node) {
-            var lastToken = context.getLastToken(node),
-                nextToken = context.getTokenAfter(node),
-                // peek few characters beyond the last token (typically the semi-colon)
-                sourceLines = context.getSource(lastToken, 0, 3);
-
-            // Some coding styles like Google uses multiple `var` statements.
-            // So if the next token is a `var` statement don't do anything.
-            if (nextToken && nextToken.type === "Keyword" &&
-                (nextToken.value === "var" || nextToken.value === "let" || nextToken.value === "const")) {
-                return;
-            }
-
-            // Next statement is not a `var`
-            if (sourceLines && !sourceLines.match(validator)) {
-                context.report(node, "Newline is " + mode + " expected after a \"var\" statement.",
-                               { identifier: node.name });
-            }
+    /**
+     * Checks that a blank line exists after a variable declaration when mode is
+     * set to "always", or checks that there is no blank line when mode is set
+     * to "never"
+     * @private
+     * @param {ASTNode} node - `VariableDeclaration` node to test
+     * @returns {void}
+     */
+    function checkForBlankLine(node) {
+        var lastToken = context.getLastToken(node),
+            nextToken = context.getTokenAfter(node),
+            hasBlankLine;
+
+        // Ignore if there is no following statement
+        if (!nextToken) {
+            return;
+        }
+
+        // Some coding styles like Google uses multiple `var` statements.
+        // So if the next token is a `var` statement don't do anything.
+        if (nextToken.type === "Keyword" && isVar(nextToken.value)) {
+            return;
+        }
+
+        // Next statement is not a `var`
+        hasBlankLine = nextToken.loc.start.line > lastToken.loc.end.line + 1;
+
+        if (mode === "never" && hasBlankLine) {
+            context.report(node, NEVER_MESSAGE, { identifier: node.name });
         }
+
+        if (mode === "always" && !hasBlankLine) {
+            context.report(node, ALWAYS_MESSAGE, { identifier: node.name });
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    return {
+        "VariableDeclaration:exit": checkForBlankLine
     };
```

Comment:
Yea, it would seem so! I updated this commit on my fork but I am sad to say I can't seem to get the actual Pull Request to update with the new change here. Thoughts?